### PR TITLE
Merge changes for Ranch into Develop

### PR DESCRIPTION
**Pull request by e-lo**
### 1. Add CRS_alt

* @i-am-sijia please add what this is doing/why or attach an issue. Updated descriptions in #302 

### 2. Adds/updates geometry

* If a node x/y is changed or a new node is added, update the geometry field for geopandas
* Added as #297

### 3. Change default for `UNIQUE_SHAPE_ID` field

* changed from `shape_id` to `id` because shapes out of Ranch has unique `id` from SharedStreets, no need to create a `shape_id`. ( @i-am-sijia please fill in reason)
* Fixed in #296

### 4. Updated `base_scenario` instantiation

* Translate `base_scenario` to dictionary if  `base_scenario` attribute is of type `Scenario` so that it is consistent type as it is read in.
* Fixed in #299

### 5. Added to managed lane required attributes

* Added "assign_group",  "county",  "mpo" as required fields for managed lanes. 
* Fixed in #296

### CODE REVIEW

**Comment by e-lo** [comment]
> 3. Change default for UNIQUE_SHAPE_ID field
> changed from shape_id to id because... ( @i-am-sijia please fill in reason)

I think this is moot given #296?

**Review by e-lo — COMMENTED** [review]
See inline comments

**Comment by e-lo** [comment]
> 5. Added to managed lane required attributes
> Added "assign_group", "county", "mpo" as required fields for managed lanes.

I think this is also taken care of by the flexibility in #296?

File: network_wrangler/scenario.py [hunk 1]
```diff
@@ -96,6 +96,17 @@ def __init__(self, base_scenario: dict, project_cards: [ProjectCard] = None):
         self.road_net = None
         self.transit_net = None
 
+        if type(base_scenario) is dict:
```
Comment (e-lo) [inline network_wrangler/scenario.py:99]: technically it *is* supposed to be a dict per typehint?  but we can address it

**Comment by e-lo** [comment]
@i-am-sijia I think this branch is now moot based on individual PRs to `develop` (as noted) with the exception of the alt_CRS - explanation of what it is?

File: network_wrangler/roadwaynetwork.py [hunk 2]
```diff
@@ -1616,7 +1620,18 @@ def _add_dict_to_df(df, new_dict):
                     raise ValueError(msg)
 
             for node in nodes:
+                node["new_node"] = 1
                 self.nodes_df = _add_dict_to_df(self.nodes_df, node)
+            
+            # add geometry for new nodes
+            self.nodes_df["geometry"] = self.nodes_df.apply(
```
[the remark this answers is not in the document]
Comment (i-am-sijia) [reply network_wrangler/roadwaynetwork.py:1627]: RE 1: This chunk of codes is for adding new links and new nodes.  I don't think Lasso/Wrangler currently support node changes.
RE 2: I was following the implementation of new links. https://github.com/wsp-sag/network_wrangler/blob/34580a0f853eebf9d41d45b7c31d6c8c3ad0743c/network_wrangler/roadwaynetwork.py#L1677-L1696

File: network_wrangler/roadwaynetwork.py [hunk 3]
```diff
@@ -148,6 +148,7 @@ class RoadwayNetwork(object):
 
     # CRS = "+proj=longlat +ellps=WGS84 +datum=WGS84 +no_defs"
     CRS = 4326  # "EPSG:4326"
+    CRS_alt=4269
```
[the remark this answers is not in the document]
Comment (i-am-sijia) [reply network_wrangler/roadwaynetwork.py:151]: Perhaps we can name it `alternative_standard_crs`. Added details in #302

**Comment by i-am-sijia** [comment]
I added descriptions in #302 regarding the CRS issue.

File: network_wrangler/roadwaynetwork.py [hunk 4]
```diff
@@ -187,6 +188,9 @@ class RoadwayNetwork(object):
         "roadway",
         "length",
         "segment_id",
+        "assign_group",
```
[the remark this answers is not in the document]
Comment (i-am-sijia) [reply network_wrangler/roadwaynetwork.py:191]: Agreed.

**Comment by e-lo** [comment]
@i-am-sijia I believe this is now all moot and we can close this PR? Please check.

**Comment by e-lo** [comment]
This is a dupe of functionality already in v1.

**Pull request closed without merging**